lib/logstash/config/config_ast.rb

```diff
@@ -211,13 +216,7 @@ def compile
         return "start_input(#{variable_name})"
       when "filter"
         return <<-CODE
-          events = events.flat_map do |event|
-            next [] if event.cancelled?
-
-            new_events = []
-            #{variable_name}.filter(event){|new_event| new_events << new_event}
-            event.cancelled? ? new_events : new_events.unshift(event)
-          end
+          events = #{variable_name}.multi_filter(events)
         CODE
```

Comment:
I agree to remove it, if each plugin emits events and not array.
We should have a clearer deprecated policies for third parties plugins. cc @suyograo